lib/coach/handler.rb

```diff
@@ -7,25 +7,27 @@ module Coach
   class Handler
     STATUS_CODE_FOR_EXCEPTIONS = 0
 
+    attr_reader :name
+
     def initialize(middleware, config = {})
-      @root_item = MiddlewareItem.new(middleware, config)
-      validate!
-    rescue Coach::Errors::MiddlewareDependencyNotMet => e
-      # Remove noise of validation stack trace, reset to the handler callsite
-      e.backtrace.clear.concat(Thread.current.backtrace)
-      raise e
+      @config = config
+      if middleware.is_a?(String)
+        @name = middleware
+      else
+        @middleware = middleware
+        @name = middleware.name
+        root_item
```

Comment:
I think it would be worth adding a comment here to clarify that we're making this call to trigger validation